File: testbed/test/cruise/statemachine/test/SimpleStateMachineTest.java
package cruise.statemachine.test;

import org.junit.*;

public class SimpleStateMachineTest
{

  @Test
  public void OneStateNoEvents()
  {
    CourseB course = new CourseB();
    Assert.assertEquals(CourseB.Status.Open, course.getStatus());
  }
  
  @Test
  public void StateMachineWithNegativeNumberGuard()
  {
    StateMachineWithNegativeNumberGuard sm = new StateMachineWithNegativeNumberGuard();
    Assert.assertEquals(StateMachineWithNegativeNumberGuard.Status.On, sm.getStatus());
    sm.turnOff(-1);
    Assert.assertEquals(StateMachineWithNegativeNumberGuard.Status.On, sm.getStatus());
    sm.turnOff(0);
    Assert.assertEquals(StateMachineWithNegativeNumberGuard.Status.Off, sm.getStatus());
  }
  
  @Test
  public void StateMachineWithStringComparisonGuard()
  {
    StateMachineWithStringComparisonGuard sm = new StateMachineWithStringComparisonGuard();
    Assert.assertEquals(StateMachineWithStringComparisonGuard.CommandControl.IDLE, sm.getCommandControl());
    sm.execute();
    Assert.assertEquals(StateMachineWithStringComparisonGuard.CommandControl.CHECK_AVAILABILITY, sm.getCommandControl());
	sm.execute();
    Assert.assertEquals(StateMachineWithStringComparisonGuard.CommandControl.CHECK_AVAILABILITY, sm.getCommandControl());
	sm.setCmdString("invalid");
	sm.execute();
	Assert.assertEquals(StateMachineWithStringComparisonGuard.CommandControl.COMMAND_VALIDITY, sm.getCommandControl());
	sm.execute();
	Assert.assertEquals(StateMachineWithStringComparisonGuard.CommandControl.COMMAND_VALIDITY, sm.getCommandControl());
	sm.setCmdString("valid");
	sm.execute();
	Assert.assertEquals(StateMachineWithStringComparisonGuard.CommandControl.EXECUTION, sm.getCommandControl());
  }
}

File: testbed_python/test/statemachine/simple_state_machine_test.py
import unittest

# Test classes are imported dynamically
# Must be generated into ImportModules' namespace then imported
from ImportModules import importModules

importModules(
    ["CourseB", "StateMachineWithNegativeNumberGuard"],
    ["statemachine", "test"],
)
from ImportModules import *


class SimpleStateMachineTest(unittest.TestCase):
    def OneStateNoEvents(self):
        course = CourseB.CourseB()
        self.assertEquals(CourseB.CourseB.Status.Open, course.getStatus())

    def StateMachineWithNegativeNumberGuard(self):
        sm = StateMachineWithNegativeNumberGuard.StateMachineWithNegativeNumberGuard()
        self.assertEquals(
            StateMachineWithNegativeNumberGuard.StateMachineWithNegativeNumberGuard.Status.On,
            sm.getStatus(),
        )
        sm.turnOff(-1)
        self.assertEquals(
            StateMachineWithNegativeNumberGuard.StateMachineWithNegativeNumberGuard.Status.On,
            sm.getStatus(),
        )
        sm.turnOff(0)
        self.assertEquals(
            StateMachineWithNegativeNumberGuard.StateMachineWithNegativeNumberGuard.Status.Off,
            sm.getStatus(),
        )
